keylime/tpm2.py

```diff
@@ -568,11 +599,12 @@ def __create_aik(self, activate, asym_alg=None, hash_alg=None, sign_alg=None):
                 'opw': owner_pw,
                 'apw': aik_pw
             }
-            if legacy_tools:
+
+            if tools_version == "3.2":
                 command = "tpm2_getpubak -E {ekhandle} -k 0x81010008 -g {asymalg} -D {hashalg} -s {signalg} -f {akpubfile} -e {epw} -P {apw} -o {opw}".format(**cmdargs)
-            else:
-                command = "tpm2_createak -C {ekhandle} -k - -G {asymalg} -D {hashalg} -s {signalg} -p {akpubfile} -f pem -e {epw} -P {apw} -o {opw}".format(**cmdargs)
-            retDict = self.__run(command, outputpaths=akpubfile.name)
+            elif tools_version == "4.0":
+                command = "tpm2_createak -Q -C {ekhandle} -c ak.ctx -G {asymalg} -g {hashalg} -s {signalg} -u {akpubfile} -f pem -n ak.name -p {apw} -P {epw}".format(**cmdargs)
```

Comment:
I wouldn't recommend using `-Q` here, since STDOUT is how we read `akname` in below.  You can also remove `-n ak.name` if you do that.